### PR TITLE
Adding a new action for running infracost

### PR DESCRIPTION
**Pull request by kimgri**
Implementing infracost action, used from 
**plan-for-apply-on-pr
plan-for-apply-on-pr-comment
destroy-on-pr-merge**

### CODE REVIEW

File: infracost/action.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    description: Terragrunt exit code, e.g. '0' for success
+    value: ${{ steps.this.outputs.exit-code }}
+
+runs:
+  using: composite
+  steps: 
+    - id: this
+      shell: bash
+      run: |
+        terragrunt show -json ${{ runner.temp }}/plan.tfplan > ${{ runner.temp }}/plan.json
```
Comment (hluotola-paf) [inline infracost/action.yml:35]: Path to saved binary plan cloud be input (it can default to this value).

File: infracost/action.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+outputs:
+  output:
+    description: infracost output (diff)
+    value: ${{ steps.this.outputs.output }}
+  summary: 
+    description: Short summary containing old monthly + new monthly
+    value: ${{ steps.this.outputs.summary }}
+  exit-code:
+    description: Terragrunt exit code, e.g. '0' for success
+    value: ${{ steps.this.outputs.exit-code }}
```
Comment (hluotola-paf) [inline infracost/action.yml:27]: Just remove

File: terraform/pr-comment/action.yml [hunk 3]
```diff
@@ -42,7 +48,15 @@ runs:
...
           echo "### Terraform \`plan\` ($1)"
-          details terraform closed "$2"
+          summary_title="Show output"
+          details terraform closed "$2" "$summary_title"
+          echo ""
+
+          summary_title="Show Price Estimation"
+          echo "### Cost Estimation Plan ($1)" 
+          echo "**Cost Summary:**" "$5"
+          details terraform closed "$4" "$summary_title"
```
Comment (hluotola-paf) [inline terraform/pr-comment/action.yml:58]: Change codeblock syntax  from `terraform` to something else?

**Review by hluotola-paf — CHANGES_REQUESTED** [review]

File: infracost/action.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
+name: infracost
+description: >
+  Runs infracost to get a cost estimate.
+  
+inputs:
+  WORKING_DIRECTORY:
```
Comment (jpalomaki) [inline infracost/action.yml:6]: Please use `kebab-case` for inputs

File: infracost/action.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    required: true
+    default: ${{ runner.temp }}
+  max-output-length:
+    description: Maximum output length (combined stdout + stderr), in characters
+    required: false
+    default: 65000
+  on-truncate-message:
+    description: Message to append to output, if it gets truncated due to exceeding max length
+    required: false
+    default: "\n[output truncated, see workflow logs for complete output]"
```
Comment (jpalomaki) [inline infracost/action.yml:17]: Not used?

File: infracost/action.yml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+runs:
+  using: composite
+  steps: 
+    - id: this
+      shell: bash
+      env:
+        MAX_OUTPUT_LENGTH: ${{ inputs.max-output-length }}
+        ON_TRUNCATE_MSG:   ${{ inputs.on-truncate-message }}
+        WORKING_DIRECTORY: ${{ inputs.working-directory }}
```
Comment (jpalomaki) [inline infracost/action.yml:35]: Shouldn't this just be set on the run step using `working-directory`? Then you could just say e.g.
```
terragrunt show -json plan.tfplan > plan.json
```

File: infracost/action.yml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+      run: |
+     
+        terragrunt show -json  $WORKING_DIRECTORY/plan.tfplan > $WORKING_DIRECTORY/plan.json
+        infracost configure set currency EUR
```
Comment (jpalomaki) [inline infracost/action.yml:39]: Prefer environment variables for config: https://www.infracost.io/docs/features/environment_variables/

File: infracost/action.yml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    required: false
+    default: "\n[output truncated, see workflow logs for complete output]"
+
+outputs:
+  output:
+    description: infracost output (diff)
+    value: ${{ steps.this.outputs.output }}
+  summary: 
+    description: Short summary containing old monthly + new monthly
+    value: ${{ steps.this.outputs.summary }}
```
Comment (jpalomaki) [inline infracost/action.yml:25]: Whats the difference between output and summary?

File: infracost/action.yml [hunk 9]
```diff
@@ -0,0 +1,49 @@
+name: infracost
+description: >
+  Runs infracost to get a cost estimate.
```
Comment (jpalomaki) [inline infracost/action.yml:3]: No need to use `>` here since its just one line. Just do
```
description: Runs infracost to get a cost estimate
```

**Comment by jpalomaki** [comment]
IMO the action should be called `run-infracost` (so it rhymes with https://github.com/pafcloud/terraform-actions/tree/master/run-terragrunt)

File: infracost/action.yml (hunk 4 above)
Comment (kimgri) [reply infracost/action.yml:6]: kebabcase caused syntax errors, (initially it was $working-directory) in other actions WORKING_DIRECTORY was also used.

Comment (jpalomaki) [reply infracost/action.yml:6]: Kebab case will just fine (and is a convention), see e.g. https://github.com/pafcloud/terraform-actions/blob/master/run-terragrunt/action.yml#L12

File: infracost/action.yml (hunk 8 above)
Comment (kimgri) [reply infracost/action.yml:17]: Output is the raw output, meaning all of the infracost data.
Summary is just: "Amount: Current -> New : Difference: Current - New "

File: run-infracost/action.yml [hunk 10]
```diff
@@ -0,0 +1,38 @@
+name: infracost
```
Comment (jpalomaki) [inline run-infracost/action.yml:1]:
```suggestion
name: run-infracost
```

File: run-infracost/action.yml [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+inputs:
+  plan-file:
+    description: Path to saved terraform plan file
+    required: true
+
+  working-directory:
+    description: Working directory, in which the infracost command is run
+    required: true
+    default: ${{ runner.temp }}
```
Comment (jpalomaki) [inline run-infracost/action.yml:12]: Remove default value (I guess `terragrunt.hcl` is never in there)?

File: run-infracost/action.yml [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+description: Runs infracost to get a cost estimate.
+
+inputs:
+  plan-file:
+    description: Path to saved terraform plan file
+    required: true
+  working-directory:
+    description: Working directory, in which the infracost command is run
+    required: true
+  infracost-api-point:
```
Comment (jpalomaki) [inline run-infracost/action.yml:11]: Should this say _endpoint_, rather than _point_?

File: run-infracost/action.yml [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+      env:
+        INFRACOST_CURRENCY: EUR
+        INFRACOST_CURRENCY_FORMAT: "EUR: 1.234,56€"
+        INFRACOST_PRICING_API_ENDPOINT: ${{ inputs.infracost-api-point }}
+        PLAN_FILE: ${{ inputs.plan-file }}
+        RESOURCE_USE: ${{ steps.resolve-usage-file.outputs.resource-use }}
+      run: |
+        tmp_dir=$(mktemp -d $RUNNER_TEMP/infracost.XXXXXXXXXX)
+        terragrunt show -json $PLAN_FILE > $tmp_dir/plan.json
+        infracost diff --path $tmp_dir/plan.json --format=json --usage-file $RUNNER_TEMP/result.yml >> $tmp_dir/infracost.json
```
Comment (jpalomaki) [inline run-infracost/action.yml:103]: `rersult.yml` sounds strange for usage file name

**Pull request merged**